nilearn/maskers/nifti_labels_masker.py

```diff
@@ -243,7 +243,7 @@ def _reporting(self):
             # previously fitted with no func image and is re-fitted
             if 'warning_message' in self._report_content:
                 self._report_content['warning_message'] = None
-            labels_image = image.load_img(labels_image, dtype=int)
+            labels_image = image.load_img(labels_image, dtype='int32')
```

Comment:
Nibabel will now require you to specify an integer size when running `set_data_dtype()` on a `Nifti1Image`. I chose `int32` to be conservative, but for a label image `int16` is probably more appropriate. Let me know what you think.

Reply:
the `load_img` call shouldn't be necessary as this is always an in-memory Nifti1Image, and rather than the reporting code, the code that creates the `labels_img_` and `_resampled_labels_img_` should be responsible for making sure it has a sensible (ie int16 or int32) dtype. we also discussed adding a check for int images in `load_img`. but I think that is outside the scope of adding the pre-releases CI job. I think the conservative int32 choice is perfectly reasonable so I suggest we keep the choice you made